cvprac/cvp_api.py

```diff
@@ -0,0 +1,500 @@
+        return data
+
+    def get_logs_by_id(self, task_id, start=0, end=0):
+        ''' Returns the log entries for the task with the specified TaskId.
+
+            Args:
+                task_id (int): CVP task identifier
+                start (int): The first log entry to return.
+                end (int): The last log entry to return.  Default is 0 which
+                    means to return all log entries.  Can be a large number to
+                    indicate the last log entry.
+
+            Returns:
+                task (dict): The CVP log for the associated Id.  Returns None
+                    if the task_id was invalid.
+        '''
+        self.log.debug('get_log_by_id: task_id: %s' % task_id)
+        return self.clnt.get('/task/getLogsById.do?id=%s&queryparam='
+                             '&startIndex=%d&endIndex=%d' %
+                             (task_id, start, end))
+
+    def add_note_to_task(self, task_id, note):
+        ''' Add notes to the task.
+
+            Args:
+                task_id (str): Task ID
+                note (str): Note to add to the task
+        '''
+        self.log.debug('add_note_to_task: task_id: %s note: %s' %
+                       (task_id, note))
+        data = {'workOrderId' : task_id, 'note' : note}
+        self.clnt.post('/task/addNoteToTask.do', data)
+
+    def execute_task(self, task_id):
+        ''' Execute the task
+
+            Args:
+                task_id (str): Task ID
+        '''
+        self.log.debug('execute_task: task_id: %s' % task_id)
+        data = {'data' : [task_id]}
+        self.clnt.post('/task/executeTask.do', data)
+
+    def cancel_task(self, task_id):
+        ''' Cancel the task
+
+            Args:
+                task_id (str): Task ID
+        '''
+        self.log.debug('cancel_task: task_id: %s' % task_id)
+        data = {'data' : [task_id]}
+        self.clnt.post('/task/cancelTask.do', data)
+
+    def get_configlet_by_name(self, name):
+        ''' Returns the configlet with the specified name
+
+            Args:
+                name (str): Name of the configlet.  Can contain spaces.
+
+            Returns:
+                configlet (dict): The configlet dict.
+        '''
+        self.log.debug('get_configlets_by_name: name: %s' % name)
+        return self.clnt.get('/configlet/getConfigletByName.do?name=%s' % name)
+
+    def get_configlet_history(self, key):
+        ''' Returns the configlet history.
+
+            Args:
+                key (str): Key for the configlet.
+
+            Returns:
+                history (dict): The configlet dict with the changes from
+                    most recent to oldest.
+        '''
+        self.log.debug('get_configlets_history: key: %s' % key)
+        return self.clnt.get('/configlet/getConfigletHistory.do?configletId='
+                             '%s&queryparam=&startIndex=0&endIndex=0' % key)
+
+    def get_inventory(self):
```

Comment:
Would there be any reason to implement the start and end values as optional rather than fixed at 0?